### PR TITLE
add authprovider owncloudsql

### PR DESCRIPTION
**Pull request by butonic**
I added an authprovider that can be configured to authenticate against an owncloud classic mysql database. It verifies the password from the oc_users table. I am using this for my personal migration / parralel setup.

Tested using sqlite and in production using this toml file for reva:
```
[shared]
jwt_secret = "Pive-Fumkiu4"

[grpc]
address = "0.0.0.0:9999"

[grpc.services.authprovider]
auth_manager = "owncloudsql"

[grpc.services.authprovider.auth_managers.owncloudsql]
dbusername = "owncloud"
dbpassword = "secret"
dbhost = "mysql"
dbport = 3306
dbname = "owncloud"
idp = "https://idp.example.org"
#nobody = 99
#legacy_salt = ""
join_username = true
join_ownclouduuid = false
```

Then running grpcurl like this:
```
$ grpcurl -d '{"client_id": "test", "client_secret": "secret"}' -import-path ./cs3apis/ -proto ./cs3apis/cs3/auth/provider/v1beta1/provider_api.proto -plaintext localhost:9999 cs3.auth.provider.v1beta1.ProviderAPI/Authenticate
{
  "status": {
    "code": "CODE_OK",
    "trace": "00000000000000000000000000000000"
  },
  "user": {
    "id": {
      "idp": "https://idp.example.org",
      "opaqueId": "test",
      "type": "USER_TYPE_PRIMARY"
    },
    "mail": "test@example.org",
    "displayName": "test"
  },
  "tokenScope": {
    "user": {
      "resource": {
        "decoder": "json",
        "value": "***redacted***"
      },
      "role": "ROLE_OWNER"
    }
  }
}
```

I am only using this to test basic auth for some routes that the ocis proxy forwards to reva ... but I'd like to forward auth requests to the gateway, and then I could directly authenticate without having to go throug an ldap proxy.

### CODE REVIEW

File: pkg/auth/manager/owncloudsql/accounts/accounts.go [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+// Accounts represents oc10-style Accounts
+type Accounts struct {
+	driver                                     string
+	db                                         *sql.DB
+	joinUsername, joinUUID, enableMedialSearch bool
+	selectSQL                                  string
+}
+
+// NewMysql returns a new Cache instance connecting to a MySQL database
```
Comment (C0rby) [inline pkg/auth/manager/owncloudsql/accounts/accounts.go:39]: > `new Cache instance`

Was `Cache` intentional? If so, why is it a cache?

File: pkg/auth/manager/owncloudsql/accounts/accounts.go [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+	err = sqldb.Ping()
+	if err != nil {
+		return nil, errors.Wrap(err, "error connecting to the database")
+	}
+
+	return New("mysql", sqldb, joinUsername, joinUUID, enableMedialSearch)
+}
+
+// New returns a new Cache instance connecting to the given sql.DB
```
Comment (C0rby) [inline pkg/auth/manager/owncloudsql/accounts/accounts.go:57]: Here too `Cache`

File: pkg/auth/manager/owncloudsql/owncloudsql.go [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+	"github.com/cs3org/reva/pkg/appctx"
+	"github.com/cs3org/reva/pkg/auth"
+	"github.com/cs3org/reva/pkg/auth/manager/owncloudsql/accounts"
+	"github.com/cs3org/reva/pkg/auth/manager/registry"
+	"github.com/cs3org/reva/pkg/auth/scope"
+	"github.com/cs3org/reva/pkg/errtypes"
+	"github.com/mitchellh/mapstructure"
+	"github.com/pkg/errors"
+
+	// Provides mysql drivers
```
Comment (refs) [inline pkg/auth/manager/owncloudsql/owncloudsql.go:42]: unnecessary comment

File: pkg/auth/manager/owncloudsql/owncloudsql.go [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return mgr, nil
+}
+
+// new returns a dummy auth manager for testing
+func new(m map[string]interface{}) (*manager, error) {
```
Comment (refs) [inline pkg/auth/manager/owncloudsql/owncloudsql.go:91]: if this is for testing could it be moved to a test file?

**Review by refs — APPROVED** [review]
👍 Thanks for the tests!

A few remarks:

- is the binary file `test.sql` included by mistake?
- if the use of queries becomes more prominent we should consider using a [query builder](https://github.com/doug-martin/goqu)

**Review by C0rby — DISMISSED** [review]

**Comment by butonic** [comment]
@labkode @ishank011 rebased, added CODEOWNERS line, good to merge

**Review by labkode — APPROVED** [review]

**Pull request merged**